code/krepel/container/array.d

```diff
@@ -227,6 +236,25 @@ struct Array(T)
     Data = Data[0 .. $ - CountToRemove];
   }
 
+  void Insert(IndexType)(IndexType Where, const(ElementType[]) ToInsert)
+  {
+    assert(Where >= 0 && Where <= Count);
+    auto DataToMove = Data[Where .. $];
+    auto ExpandedArea = ExpandUninitialized(ToInsert.length);
+    if (Where == Count)
+    {
+      ExpandedArea[] = ToInsert[];
+    }
+    else
+    {
+      for(long Index = Count - 1; Index >= Where + ToInsert.length; Index--)
+      {
+        Data[Index] = Data[Index - ToInsert.length];
+      }
+      Data[Where .. Where + ToInsert.length] = ToInsert[];
+    }
+  }
+
```

Comment:
We should add an overload for `Insert` that takes a single element. Also, like `PushBack`, I would take any kind of type to insert as long as it is compatible to `ElementType`. But we can do that later.